src/main/java/de/eldoria/pickmeup/listener/CarryListener.java

```diff
@@ -92,57 +92,113 @@ public void onEntityInteract(PlayerInteractAtEntityEvent event) {
     @EventHandler
     public void onSneak(PlayerToggleSneakEvent event) {
         Player player = event.getPlayer();
-        if (!mountStates.containsKey(player.getUniqueId())) {
+        UUID playerUUID = player.getUniqueId();
+        MountState currentState = mountStates.get(playerUUID);
+
+        if (currentState == null) {
             unmountAll(player);
             return;
         }
 
-        MountState mountState = mountStates.get(player.getUniqueId());
+        if (!event.isSneaking()) {
+            handleSneakRelease(player, playerUUID, currentState);
+        } else {
+            handleSneakPress(player, playerUUID, currentState);
+        }
+    }
 
-        if (!event.isSneaking() && mountState == MountState.SNEAK_MOUNT) {
-            mountStates.put(player.getUniqueId(), MountState.WALKING);
-            return;
+    private void handleSneakRelease(Player player, UUID playerUUID, MountState currentState) {
+        switch (currentState) {
+            case SNEAK_MOUNT:
+                mountStates.put(playerUUID, MountState.WALKING);
+                break;
+            case SNEAK_THROW:
+                if (!throwBarHandler.isRegistered(player)) {
+                    unmountAll(player);
+                } else {
+                    performThrow(player, playerUUID);
+                }
```

Comment:
Better code readability. If you do if else avoid negations, since those are easily overlooked
```suggestion
                if (throwBarHandler.isRegistered(player)) {
                    performThrow(player, playerUUID);
                } else {
                    unmountAll(player);
                }
```